expression/builtin_compare.go

```diff
@@ -339,6 +342,28 @@ func (b *builtinCoalesceDurationSig) evalDuration(row chunk.Row) (res types.Dura
 	return res, isNull, err
 }
 
+// builtinCoalesceJsonSig is buitin function coalesce signature which return type time
+// See http://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_coalesce
+type builtinCoalesceJsonSig struct {
```

Comment:
lint tool suggests "type builtinCoalesceJsonSig should be builtinCoalesceJSONSig"